### PR TITLE
testthat workflow using integration tests

### PR DESCRIPTION
**Pull request by alandipert**
This adds a new `testthat.yml` workflow for running `testthat`/unit/integration tests on any application containing a `tests/testthat.R` file. `testthat.R` is sourced via `shiny::runTests()`, so code in apps with an `R/` directory (like module code) is loaded before any test-related code is executed.

This PR also adds a new `shinycoreci::trigger()` function that can be used to programmatically trigger workflows from the R console.

The testthat workflow won't be exercised until https://github.com/rstudio/shinycoreci-apps/pull/2 adds apps to test, but adding it beforehand should be fine (it will do nothing).

### CODE REVIEW

File: R/test_testthat.R [hunk 1]
```diff
@@ -0,0 +1,34 @@
+#' Test testthat
+#'
+#' @param dir Name of directory to look for shiny apps, and to save platform
+#'   information.
+#' @param apps Character vector of shiny applications
+#' @param suffix Suffix to use for sysinfo file
+#' @export
+test_testthat <- function(
+  dir = ".",
```
Comment (schloerke) [inline R/test_testthat.R]: For consistency... set default to `"apps"`

File: R/test_testthat.R [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  appdirs <- file.path(dir, apps)
+
+  fail_apps <- character(0)
+  for (appdir in appdirs) {
+    message("Testing ", appdir)
+    tryCatch(
+      shiny::runTests(appDir = appdir, filter = "^testthat.R$"),
+      error = function(failed) {
+        message("Failed: ", failed$message)
+        fail_apps <<- c(fail_apps, basename(appdir))
```
Comment (schloerke) [inline R/test_testthat.R]: Up for debate....

Store a `fail_apps <- data.frame(app = character(0), error = character(0), stringsAsFactors = FALSE)` before the for-loop and append using `fail_apps[nrow(fail_apps) + 1, ] <<- list(app = appdir, message = failed$message)`.

Final check against `nrow(fail_apps)` at the end.

Comment (alandipert) [reply R/test_testthat.R]: To append faster you mean? I'm not against it, but I don't see a particular need at the moment. A potential worthwhile future improvement that comes to mind is to move error aggregation logic into a separate set of shared helper functions.

Comment (schloerke) [reply R/test_testthat.R]: To append faster: no.  Only done to save all errors and app combos in a data.frame that can be inspected at the end.

I'm ok with waiting until it's standardized.

**Pull request merged**